src/controllers/accounts/accounts.ts

```diff
@@ -216,6 +216,18 @@ export class AccountsController extends EventEmitter {
     await this.#storage.set('accounts', this.accounts)
   }
 
+  async reorderAccounts(reordered: Account[]) {
+    const reorderedAddrSet = new Set(reordered.map((acc) => getAddress(acc.addr)))
+
+    const remainingAccounts = this.accounts.filter(
+      (acc) => !reorderedAddrSet.has(getAddress(acc.addr))
+    )
+
+    this.accounts = getUniqueAccountsArray([...reordered, ...remainingAccounts])
```

Comment:
As far as I understand this code serves as guarantee that there will be no removed accounts if `reorderAccounts` is not called with all accounts. What concerns me is that we spread `remainingAccounts` after `reordered`. Let's discuss this offline